core/keyboard_nav/navigation.js

```diff
@@ -662,6 +662,60 @@ Blockly.navigation.handleEnterForWS = function() {
 /** Helper Functions **/
 /**********************/
 
+/**
+ * Finds the source block of the location on a given node.
+ * @param {Blockly.ASTNode} node The node to find the source block on.
+ * @return {Blockly.Block} The source block of the location on the given node,
+ * or null if the node is of type workspace.
+ * @private
+ */
+Blockly.navigation.getSourceBlock_ = function(node) {
+  if (!node) {
+    return null;
+  }
+  if (node.getType() === Blockly.ASTNode.types.BLOCK) {
+    return node.getLocation();
+  } else if (node.getType() === Blockly.ASTNode.types.WORKSPACE) {
+    return null;
+  } else {
+    return node.getLocation().getSourceBlock();
+  }
+};
+
+/**
+ * Before a block is deleted move the cursor to the appropriate position.
+ * @param {!Blockly.Block} deletedBlock The block that is being deleted.
+ * @package
+ */
+Blockly.navigation.moveCursorOnBlockDelete = function(deletedBlock) {
+  var cursor = Blockly.navigation.cursor_;
+  if (cursor) {
+    var curNode = cursor.getCurNode();
+    var block = Blockly.navigation.getSourceBlock_(curNode);
+
+    if (block === deletedBlock) {
+      // If the block has a parent move the cursor to their connection point.
+      if (block.getParent()) {
+        var topConnection = block.previousConnection
+          ? block.previousConnection : block.outputConnection;
```

Comment:
Move the ? to the end of the previous line ("When a line is broken at an operator the break comes after the symbol. ")